src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -325,11 +396,23 @@ private Automaton toAutomaton() {
     }
 
     public StringFilter convertToStringFilter() {
-        return new StringFilter(toAutomaton());
+        try {
+            return new AutomatonBackedStringFilter(toAutomaton());
+        } catch (TooComplexToDeterminizeException e) {
+            // Hard to pre-empt when this exception will occur due to the number
+            // terms and possible overlap of contents
+            // so we let it try and fall-back to Set based lookups if too
+            // complex.
+            return new TermListBackedStringFilter(includeValues, excludeValues);
+        }
```

Comment:
Even it we remove that logic, it would be good though to keep the explanation that we don't use an automaton for the term list case in order not to blow up the maximum number of states.